src/packet/ipv4.rs

```diff
@@ -76,11 +79,28 @@ pub struct Ipv4Option {
     copied: u1,
     class: u2,
     number: u5,
-    length: u8,
+    #[length_fn = "ipv4_options_length_length"]
```

Comment:
Could you add a comment here specifying that this is an optional field, rather than a vector?